lib/asset_cloud/base.rb

```diff
@@ -8,8 +8,25 @@ class IllegalPath < StandardError
 
   class Base
     cattr_accessor :logger
-
-    VALID_PATHS = /\A[a-z0-9][a-z0-9_\-\/]+([a-z0-9][\w\-\ \.\@]*\.\w{2,6})?\z/i
+    VALID_PATHS = /\A
+      (
+        ([a-z0-9])            #Filename can be a single letter
+        |                     #OR it is many and follows the below rules
+        (
+          (\.?[\w\-\@])       #It can start with a dot but it must have a following character
+          (
+            [\w\-\@]          #You can have a letter without any following conditions
+            |
+            [\ ][\w\-\@]      #If there is a space you need to have a normal letter afterward
+            |
+            [\/][\w\-\@]      #If there is a slash you need to have a normal letter afterward
+            |
+            [\/][\.][\w\-\@]  #Though a slash could be followed by a dot so long as there is a normal letter afterward
+            |
+            [\.][\w\-\@]+     #A dot must be followed by one (or more) normal letters
+          )*                  #Zero to many of these combinations.
+        )
+      )\z/ix                   #Case insensitive
```

Comment:
This helps with the explination alot, what i meant with multiple lines was more this though

here: https://github.com/Shopify/asset_cloud/blob/b55543a15079ea56e37959bd1bd0fa31efb07a6e/lib/asset_cloud.rb#L50 and here: https://github.com/Shopify/asset_cloud/blob/d1dc85937f4145f9ba314c03ce80f83cbef96c3f/lib/asset_cloud/base.rb#L159. Do something like

```ruby
def supports?(key)
  return false unless key =~ REGEX_FOR_NO_RELATIVE_PATHS
  key =~ REGEX_THAT_ENSURES_ONLY_VALID_CHARS_PRESENT &&
  key =~ REGEX_THAT_ENSURES_NO_LEADING_DOTS
end
```